src/v/iceberg/BUILD

```diff
@@ -0,0 +1,79 @@
+load("//bazel:build.bzl", "redpanda_cc_library")
+load("//bazel/thirdparty:avro.bzl", "avro_cc_library")
+
+package(default_visibility = ["//src/v/iceberg:__subpackages__"])
+
+avro_cc_library(
+    name = "manifest_file",
+    include_prefix = "iceberg",
+    namespace = "iceberg",
+    schema = "avroschemas/manifest_file.schema.json",
+)
+
+avro_cc_library(
+    name = "manifest_entry",
+    include_prefix = "iceberg",
+    namespace = "iceberg",
+    schema = "avroschemas/manifest_entry.schema.json",
+)
+
+redpanda_cc_library(
+    name = "avro_utils",
+    hdrs = [
+        "avro_utils.h",
+    ],
+    include_prefix = "iceberg",
+    deps = [
+        "//src/v/bytes:iobuf",
+        "@avro",
+    ],
+)
+
```

Comment:
We chatted about this, but thought i'd summarize here for posterity. We don't necessarily need these each to be their own library. But the thinking is that bazel will be much better about rebuilding the minimal set of things when we tell it about these finer grained dependencies.

Longer term, as other modules will depend on Iceberg, we may build a separate top-level public iceberg library. But for now, this we'll just have these fine grained libs.

Reply:
My first thought was that it's an interesting strategy since it feels like it duplicates the includes of .cc files. It makes sense to have to describe the dependencies to the build, and it seems fine for now. I do wonder if this will eventually become unwieldy 🤷

Reply:
> I do wonder if this will eventually become unwieldy

maybe @rockwotj can chime in about this?

Reply:
> I do wonder if this will eventually become unwieldy 🤷

In comparison to cmake yes possibly. Unwieldy is a scale tho, and it will be more work to think about the build system and have to update it. I think it's a good thing. It will enforce you don't have cycles and (in my opinion and experience) will also produce more organized and more thought out code.

One can always opt out of this and glob stuff together, but that leads to worse caching. Bazel does have some optimizations to improve dependency tracking even if you make large libraries (they parse headers to determine includes and some other hacky behavior they'd love to get rid of). Ultimately this is the best way to scale builds. Sometimes this will lead to people cramming more things into a header/impl pair, but I'm hoping instead people will be more thoughtful. If you had to spell out these dependencies in the build system, I doubt the cluster directory would have ended up in the state it's in with cycles etc. 

Maybe this pattern will slow development down, but I believe it will be for the better, because if you're actually thinking about the dependency tree in the first place it shouldn't be much work to type it out in the build. There are tools in the bazel ecosystem to generate BUILD files if we want to come up with common patterns instead.

TL;DR;

Yes it's possible it will become very cumbersome for spell out deps, but it's not transitive deps, it will scale the build better, and force people to think about the dependency tree. All this outweighs the alternative, but I understand how others can feel differently.

Reply:
Yea that's fair. I guess TBD whether one-per-cc-file will be the long term strategy, but it's an easy rule to remember, and it makes sense for a young library like iceberg.

+1 about being more thoughtful about dependencies/boundaries. As clearer lines form (e.g. can imagine a library for just manifest struct definitions, and another for manifest serialization) I can imagine we'll whittle down the number to represent just well-defined, commonly-used abstractions. Or not, if a large number of libraries isn't problematic.

Reply:
> Or not, if a large number of libraries isn't problematic.

I mean lots of libraries is totally fine with bazel (one of my targets at google was 16k deps (or cc_libraries) to build)